docs/rules/prefer-usememo-over-useeffect-usestate.md

```
@@ -4,70 +4,85 @@
-    items.reduce((sum, item) => sum + item.price, 0),
-  [items]);
-
-  return <div>Total: {total}</div>;
+function Component({ a, b }) {
+  const sum = useMemo(() => a + b, [a, b]);
+  return <div>{sum}</div>;
 }
 ```
 
-### ✅ Correct (With disable comment if side effects are required)
+2. **Legitimate side effects**: Using `useEffect` is correct when you perform asynchronous operations or other side effects, like fetching data.
```

markdownlint-cli2 (0.18.1)

Ordered list item prefix
Expected: 1; Actual: 2; Style: 1/1/1

(MD029, ol-prefix)